### PR TITLE
Add MacStadium to inst_partners.yaml

### PR DESCRIPTION
**Pull request by chrisburr**
@SylvainCorlay Is this data used anywhere yet?

### CODE REVIEW

File: src/inst_partners.yaml [hunk 1]
```diff
@@ -93,9 +93,17 @@ Works on Arm:
...
    Logo URL:
      - https://www.frenchtechbordeaux.com/wp-content/uploads/2019/08/LogoOVHCloud_Vertical-Blue2.png
    contribution type:
      - infrastructure
+
+MacStadium:
+   Description: MacStadium provides dedicated Mac mini hardware
+   Site URL: https://www.macstadium.com/
+   Logo URL:
+     - https://uploads-ssl.webflow.com/5ac3c046c82724970fc60918/5c019d917bba312af7553b49_MacStadium-developerlogo.png
```
Comment (beckermr) [inline src/inst_partners.yaml:107]: We'll need some agreement to use their logo. Otherwise, we should comment out or leave this out.

Comment (chrisburr) [reply src/inst_partners.yaml:107]: They've explicitly requested that we use this URL somewhere.

Comment (beckermr) [reply src/inst_partners.yaml:107]: We do not have a page yet, so we'll need to make one.

Comment (SylvainCorlay) [reply src/inst_partners.yaml:107]: yes indeed this data is not used yet.

**Pull request merged**